server/webapp/WEB-INF/rails.new/spec/controllers/api_v1/admin/pluggable_scms_controller_spec.rb

```diff
@@ -0,0 +1,310 @@
+      end
+
+      it 'should disallow anonymous users, with security enabled' do
+        enable_security
+        login_as_anonymous
+
+        expect(controller).to disallow_action(:put, :update, material_name: 'foo').with(401, 'You are not authorized to perform this action.')
+      end
+
+      it 'should disallow normal users, with security enabled' do
+        enable_security
+        login_as_user
+
+        expect(controller).to disallow_action(:put, :update, material_name: 'foo').with(401, 'You are not authorized to perform this action.')
+      end
+
+      it 'should allow admin, with security enabled' do
+        enable_security
+        login_as_admin
+
+        expect(controller).to allow_action(:put, :update)
+      end
+
+    end
+    describe 'admin' do
+      before(:each) do
+        enable_security
+        login_as_admin
+      end
+
+      it 'should deserialize scm object from given parameters' do
+        controller.stub(:check_for_stale_request).and_return(nil)
+        hash = {id: '1', name: 'material', auto_update: false, plugin_configuration: {id: 'github.pr', version: '1'}, configuration: [{"key" => 'url', "value" => 'git@github.com:foo/bar.git'}]}
+        @pluggable_scm_service.should_receive(:updatePluggableScmMaterial).with(anything, an_instance_of(SCM), anything)
+        params = {material_name: 'material', id: '1', name: 'material', auto_update: false, plugin_configuration: {id: 'github.pr', version: '1'}, configuration: [{"key" => 'url', "value" => 'git@github.com:foo/bar.git'}]}
+        put_with_api_header :update, params
+
+        expect(response.code).to eq('200')
+        real_response = actual_response
+        real_response.delete(:_links)
+        expect(real_response).to eq(hash)
+      end
+
+      it 'should not allow rename of material name' do
+        controller.stub(:check_for_stale_request).and_return(nil)
+        params = {material_name: 'foo', id: '1', name: 'material', auto_update: false, plugin_configuration: {id: 'github.pr', version: '1'}, configuration: [{"key" => 'url', "value" => 'git@github.com:foo/bar.git'}]}
+
+        put_with_api_header :update, params
+
+        expect(response.code).to eq('406')
+        expect(actual_response[:message]).to eq('Renaming of SCM material is not supported by this API.')
+
+      end
+
+      it 'should not update existing material if validations fail' do
+        controller.stub(:check_for_stale_request).and_return(nil)
+        controller.stub(:check_for_scm_rename).and_return(nil)
+        result = double('HttpLocalizedOperationResult')
+        HttpLocalizedOperationResult.stub(:new).and_return(result)
+        result.stub(:isSuccessful).and_return(false)
+        result.stub(:message).with(anything()).and_return("Update failed")
+        result.stub(:httpCode).and_return(422)
+        @pluggable_scm_service.should_receive(:updatePluggableScmMaterial).with(anything, an_instance_of(SCM), result)
```

Comment:
You've already tested that `updatePluggableScmMaterial` is called. So instead of mocks returning mocks (which is a really bad antipattern, that should be avoided like a plague), what you want to do here is to `stub` the `updatePluggableScmMaterial` —

```ruby
@pluggable_scm_service.stub(:updatePluggableScmMaterial).with(....) do |..., result|
  result.unprocessableEntity(...)
end
```